torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -464,16 +464,31 @@ Tensor prod_backward(Tensor grad, const Tensor& input, Tensor result, int64_t di
   }
 }
 
-Tensor solve_jvp(
-    const Tensor& input_primal,
-    const Tensor& input_tangent,
-    const Tensor& other_tangent,
-    const Tensor& result) {
-  bool vector_case = at::native::linalg_solve_is_vector_rhs(input_tangent, other_tangent);
-  if (vector_case) {
-    return at::linalg_solve(input_primal, other_tangent - at::matmul(input_tangent, result.unsqueeze(-1)).squeeze(-1));
+// std::function is used to be able to pass lambdas that capture
+using solve_f = std::function<Tensor(const Tensor&, const Tensor&)>;
+static Tensor generic_solve_jvp(
+  solve_f solve,
+  const Tensor& X, const Tensor& A,
```

Comment:
@albanD , I know how to eliminate the usage of `std::function` with variadics so that we could pass lambdas which do not capture. Do you think this is worth the effort to eliminate the overhead that comes with `std::function`?

Reply:
I don't think so no.